lib/integrations.rb

```diff
@@ -1,11 +1,16 @@
 # frozen_string_literal: true
 module Integrations
   class Error < StandardError
-    attr_reader :type, :message
+    attr_reader :type, :message, :response, :response_body, :response_code
 
-    def initialize(type='unknown_error', message='Unknown Error')
+    def initialize(type='unknown_error', message='Unknown Error', response)
       @type = type
       @message = message
+      if response && response.respond_to?(:body)
+        @response = response
+        @response_body = response.body
+        @response_code = response.code
```

Comment:
Hmmm... I think we use different kinds of response objects (from the OAuth, HTTParty, and I think HipChat gems). Do they all play nicely here?